component/otelcol/internal/scheduler/host.go

```diff
@@ -26,6 +23,18 @@ func NewHost(l log.Logger) *Host {
 
 var _ otelcomponent.Host = (*Host)(nil)
 
+// SetExtensions sets the extensions available from the Host. It is not valid
+// to call this after the Host is in use.
+func (h *Host) SetExtensions(extensions map[otelconfig.ComponentID]otelcomponent.Extension) {
```

Comment:
Do we need any rails around this? Is it important that the `h.Extensions` not change or the contents of the map?

Reply:
Honestly it should just be a part of the constructor instead of a method. It's mainly that I don't know if changes would necessarily get picked up by implementations of otelcol components, so it probably shouldn't be changed after you pass the host to components.

WDYT? Should I just move setting the extensions and exporters to the host?

Reply:
I moved these to the constructor using the options pattern (since the type names are _very_ long). LMK what you think :)

Reply:
I like the new way! 👍